saleor/order/models.py

```diff
@@ -295,20 +295,30 @@ def move_to_group(self, item, target_group, quantity):
         try:
             target_item = target_group.items.get(
                 product=item.product, product_name=item.product_name,
-                product_sku=item.product_sku)
+                product_sku=item.product_sku, stock=item.stock)
         except ObjectDoesNotExist:
             target_group.items.create(
                 delivery_group=target_group, product=item.product,
                 product_name=item.product_name, product_sku=item.product_sku,
                 quantity=quantity, unit_price_net=item.unit_price_net,
                 stock=item.stock,
+                stock_location=item.stock_location,
                 unit_price_gross=item.unit_price_gross)
         else:
             target_item.quantity += quantity
             target_item.save()
         item.quantity -= quantity
         self.remove_empty_groups(item)
 
+    def merge_duplicates(self, item):
```

Comment:
To improve readability, I would rename this function to `merge_duplicated_lines` and add a docstring explaining its purpose. Also I think it could be moved to the "utils" module (as well as all of these functions in OrderedItemManager, but not all of them in this PR :)).